src/couch_replicator/src/couch_replicator_auth_basic.erl

```diff
@@ -0,0 +1,52 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_replicator_auth_basic).
```

Comment:
that this module has no code in it implies the abstraction is a bit off. There's no basic auth logic here.

Reply:
The idea is it preserves the current behavior better. For example, if credentials are put in the URL directly (`http://user:pass@host/db`) that's how they'd be passed to endpoints, as opposed to extracting them and applying them as basic auth headers. Most servers should handle basic auth headers but just in case there are some hand rolled ones that only know of one way to authenticate.

Another incentive is to use this as a feature flag -- if this is the only plugin in the list, it effectively means "do the old behavior before the session auth commit".

Is it worth doing it this way or would it be better to do the full extract and reapply?

Reply:
the `http://user:pass@host/db` form is never what's sent in an http request, though. the user:pass part has to be converted to `Authentication: Basic <goop>`. That translation is happening elsewhere (in ibrowse), leaving this module empty. It's just a bit odd to call this module 'basic' as if it does basic auth, when in fact it does nothing (and basic auth happens independently).

Reply:
Oh doh. That's embarrassing. You're right of course. Don't know what I was thinking. The connection is made to the host:port and then path part is sent sent without all the auth bits in front of it. I guess the only difference then is letting ibrowse do the appending of basic auth header or not.

As for the module should it be better called couch_replicator_auth_default - basically a catch-all module that is no-op and just mean "do whatever happened before". The reason for a separate module is that in some case users might enforce not falling back basic auth. Then they wouldn't add it to the end of the plugins list. Or do you think it's still better to do the extract basic auth, and then reply dance and keep its name.